src/components/app/pageLocationKeyRaces/locationUnitedStatesLiveResults/utils.ts

```diff
@@ -0,0 +1,168 @@
+import * as Sentry from '@sentry/nextjs'
+import { isBefore, startOfDay } from 'date-fns'
+import { isNil } from 'lodash-es'
+
+import { RaceStatus } from '@/components/app/pageLocationKeyRaces/locationUnitedStatesLiveResults/liveStatusBadge'
+import {
+  CandidatesWithVote,
+  CongressDataResponse,
+  PresidentialDataWithVotingResponse,
+  RacesVotingDataResponse,
+} from '@/data/aggregations/decisionDesk/types'
+import { DTSI_PersonPoliticalAffiliationCategory } from '@/data/dtsi/generated'
+import { dtsiPersonPoliticalAffiliationCategoryAbbreviation } from '@/utils/dtsi/dtsiPersonUtils'
+import {
+  convertDTSIPersonStanceScoreToLetterGrade,
+  isAntiCrypto,
+  isProCrypto,
+} from '@/utils/dtsi/dtsiStanceScoreUtils'
+import { twNoop } from '@/utils/web/cn'
+
+export const convertDTSIStanceScoreToBgColorClass = (score: number | null | undefined) => {
+  if (isNil(score)) {
+    return twNoop('bg-[#5b616e]')
+  }
+  if (score >= 90) {
+    return twNoop('bg-green-700')
+  }
+  if (score >= 70) {
+    return twNoop('bg-[#7b8508]')
+  }
+  if (score >= 50) {
+    return twNoop('bg-yellow-600')
+  }
+
+  return twNoop('bg-red-700')
+}
+
+export const getPoliticalCategoryAbbr = (category: DTSI_PersonPoliticalAffiliationCategory) => {
+  if (!category) return ''
+  return dtsiPersonPoliticalAffiliationCategoryAbbreviation(category) || ''
+}
+
+export const getVotePercentage = (
+  candidate: CandidatesWithVote | null,
+  raceData: RacesVotingDataResponse | null,
+) => {
+  if (!candidate) return 0
+
+  const totalVotes = Math.max(
+    candidate?.estimatedVotes?.estimatedVotesMid || 0,
+    raceData?.totalVotes || 0,
+  )
+  if (isNil(totalVotes)) return 0
+
+  return candidate.votes ? +((candidate.votes / totalVotes) * 100).toFixed(2) : 0
+}
+
+export const getRaceStatus = (
+  raceData: RacesVotingDataResponse | RacesVotingDataResponse[] | null | undefined,
+): RaceStatus => {
+  if (!raceData) return 'unknown'
+
+  let calledRace: boolean = false
+  if (Array.isArray(raceData)) {
+    if (raceData.length === 0) {
+      return 'unknown'
+    }
+
+    calledRace = raceData.every(race => race.hasCalledCandidate || race.advanceCandidates)
+  } else {
+    calledRace = raceData.hasCalledCandidate
+  }
+
+  if (calledRace) {
+    return 'called'
+  }
+
+  if (!Array.isArray(raceData) && raceData.advanceCandidates) {
+    return 'runoff'
+  }
+
+  let isLive: boolean = false
+  if (Array.isArray(raceData)) {
+    isLive = raceData.some(race => +race.totalVotes > 0)
+  } else {
+    isLive = +raceData.totalVotes > 0
+  }
+  if (isLive) {
+    return 'live'
+  }
+
+  const now = new Date()
+  let raceDate: Date
+  if (Array.isArray(raceData)) {
+    raceDate = new Date(raceData?.[0].raceDate || '2024-11-05')
+  } else {
+    raceDate = new Date(raceData?.raceDate || '2024-11-05')
+  }
+  if (isBefore(startOfDay(now), startOfDay(raceDate))) {
+    return 'not-started'
+  }
+
+  return 'live'
+}
```

Comment:
This function is more complex than it needs to be, you can make it more readable dividing the logic when its array from when it's not and working with early returns.

```suggestion
export const getRaceStatus = (
  raceData: RacesVotingDataResponse | RacesVotingDataResponse[] | null | undefined,
): RaceStatus => {
  if (!raceData) return 'unknown'

  const now = new Date()

  if (Array.isArray(raceData)) {
    if (raceData.length === 0) return 'unknown'
    if (raceData.every(race => race.hasCalledCandidate || race.advanceCandidates)) return 'called'
    if (raceData.some(race => +race.totalVotes > 0)) return 'live'

    const raceDate = new Date(raceData?.[0].raceDate || '2024-11-05')
    if (isBefore(startOfDay(now), startOfDay(raceDate))) return 'not-started'
    return 'live'
  }

  if (raceData.hasCalledCandidate) return 'called'
  if (raceData.advanceCandidates) return 'runoff'
  if (+raceData.totalVotes > 0) return 'live'

  const raceDate = new Date(raceData?.raceDate || '2024-11-05')
  if (isBefore(startOfDay(now), startOfDay(raceDate))) return 'not-started'
  return 'live'
}
```